test/e2e/sagemaker/tests/test_endpoint.py

```diff
@@ -39,91 +41,174 @@ def sagemaker_client():
+    (_, config_resource) = single_variant_config
+    config_resource_name = config_resource["spec"].get("endpointConfigName", None)
+
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["ENDPOINT_NAME"] = endpoint_resource_name
+    replacements["CONFIG_NAME"] = config_resource_name
+
+    reference, spec = make_reference(
+        resource_prural=ENDPOINT_RESOURCE_PLURAL,
+        resource_name=endpoint_resource_name,
+        spec_file="endpoint_base",
+        replacements=replacements,
     )
-    model_resource = k8s.create_custom_resource(model_reference, model)
-    model_resource = k8s.wait_resource_consumed_by_controller(model_reference)
-    assert model_resource is not None
+    logging.debug(spec)
+    resource = make_resource(reference, spec)
+
+    assert resource is not None
+
+    yield (reference, resource, spec)
+
+    # Delete the k8s resource if not already deleted by tests
+    if k8s.get_resource_exists(reference):
+        k8s.delete_custom_resource(reference)
+
+
+@pytest.fixture(scope="module")
+def faulty_config(get_name_suffix, single_container_model):
+    model_resource_name = get_name_suffix + "faulty-model"
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["MODEL_NAME"] = model_resource_name
 
-    config1_reference = k8s.CustomResourceReference(
-        CRD_GROUP,
-        CRD_VERSION,
-        CONFIG_RESOURCE_PLURAL,
-        config1_resource_name,
-        namespace="default",
+    model_reference, model_spec = make_reference(
+        resource_prural=MODEL_RESOURCE_PLURAL,
+        resource_name=model_resource_name,
+        spec_file="xgboost_model",
+        replacements=replacements,
     )
-    config1_resource = k8s.create_custom_resource(config1_reference, config)
-    config1_resource = k8s.wait_resource_consumed_by_controller(config1_reference)
-    assert config1_resource is not None
-
-    config2_resource_name = random_suffix_name("2-single-variant-endpoint", 32)
-    config["metadata"]["name"] = config["spec"][
-        "endpointConfigName"
-    ] = config2_resource_name
-    logging.debug(config)
-    config2_reference = k8s.CustomResourceReference(
-        CRD_GROUP,
-        CRD_VERSION,
-        CONFIG_RESOURCE_PLURAL,
-        config2_resource_name,
-        namespace="default",
+    # https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Bucket.copy
+    # copy model data to a temp S3 location and delete it after model is created on SageMaker
+    model_bucket = REPLACEMENT_VALUES["SAGEMAKER_DATA_BUCKET"]
+    s3 = boto3.resource("s3").Bucket(model_bucket)
+    copy_source = {
+        "Bucket": model_bucket,
+        "Key": "sagemaker/model/xgboost-mnist-model.tar.gz",
+    }
+    model_destination_key = "sagemaker/model/delete/xgboost-mnist-model.tar.gz"
+    s3.copy(copy_source, model_destination_key)
+    model_spec["spec"]["primaryContainer"][
+        "modelDataURL"
+    ] = f"s3://{model_bucket}/{model_destination_key}"
```

Comment:
nit: Move into helper method? Add docstring for clarification

Reply:
moved the s3 related opr